exec/java-exec/src/test/java/org/apache/drill/exec/physical/unit/TestNullInputMiniPlan.java

```diff
@@ -323,8 +323,6 @@ public void testHashJoinLeftEmpty() throws Exception {
         .build();
 
     BatchSchema expectedSchema = new SchemaBuilder()
-        .addNullable("a", TypeProtos.MinorType.BIGINT)
-        .addNullable("b", TypeProtos.MinorType.BIGINT)
```

Comment:
I wonder about this. This test came out of the "empty batches" project which attempted to handle empty inputs. If this PR finds the need to change the tests, then we are changing semantics of how we handle schemas. Are we sure we want to make this change?

If we are changing the schema, are we saying that if we get no left input (but do get a schema), that we won't product the joined schema? This would seem to be a bug rather than a feature.

Reply:
@paul-rogers Thank you for the review. I have changed the code to make this case work. I have made changes to incorporate the earlier comments. Can you please review the changes.